enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,456 @@
+    - port: 443
+      protocol: TCP
+  podSelector:
+    matchLabels:
+      ingresscontroller.operator.openshift.io/deployment-ingresscontroller: 
+```
+
+More information in the design details section.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+3. The router will enabled/disabled, be exposed on the specified ports, on the
+   specified IPs, allowing only specified IP addresses, all according to the
+   cluster admin provided configuration. This includes the status in the
+   LoadBalancer service type for the router.
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+  expose:
+    - <IP address/hostname> # Defaults to the node's exposed IP addresses.
+```
+
+For more information check each individual section.
+
+### Implementation Details
+The default router is composed of a bunch of assets part of the MicroShift
+binary. These assets come from the rebase, copied from the original router in
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator).
+There is already a [LoadBalancer service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/assets/router/service-cloud.yaml)
+included, which MicroShift does not copy yet. Depending on the options
+configured in MicroShift, this resource will need further customization done
+by MicroShift's start process.
+
+The rest of the changes imply only additions. The LoadBalancer controller needs
+an expansion on its capabilities to include more IPs in the status and log any
+changes to services. All the logic to control NetworkPolicy resources is also
+new.
+
+#### How config options change manifests
+Each of the configuration options described above has a direct effect on the
+manifests that MicroShift will apply after starting.
+
+The `router.status` option will drive whether the router `Deployment` and the
+`Service` get created.
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+```
+
+The `router.policy.ports.http` and `router.policy.ports.https` will determine,
+within the `Deployment` and the `Service`, which ports get configured to be
+exposed.
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+The `router.policy.expose` option, which is a list, will configure the
+`status.loadBalancer` field in the `Service`. Ovnk will pick up this field to
+configure the iptables rules.
+```yaml
+ingress:
+  expose:
+    - <IP address/hostname> # Defaults to the node's exposed IP addresses.
```

Comment:
Somehow ovnk does not create iptables rules if its not an IP address, so the controller will need to take care of turning hostnames into addresses (also poll DNS from time to time to see if the IP changed).
If we decide against doing that translation then we need to take care of iptables rules in the controller, instead of ovnk. This would make the feature independent from ovnk.
A user would express that by adding all the externally available IPs/hostnames in the config file. This is also the default behavior.